packages/core/integration-tests/test/server.js

```diff
@@ -118,17 +119,22 @@ describe('server', function() {
 
     await b.run();
 
-    b.reporterRunner.report({
-      type: 'buildFailure',
-      error: new Error('This is a server test error')
-    });
+    let priorContent = await fs.readFile(entry);
+    await fs.writeFile(entry, 'syntax\\error');
+
+    // TODO: Programatically wait for an error to occur, or time out. Right now,
+    //       Parcel does not expose this without creating a reporter that must
+    //       be loaded from disk.
```

Comment:
One solution to this would be to accept reporter objects as part of `new Parcel()`, or to implement emitter(s) that report information like in Parcel 1.